compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -1167,22 +1168,49 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                 }),
                                 Applicability::MaybeIncorrect,
                             );
-                            suggested = true;
+                            return;
                         }
                         _ => {}
                     }
                 }
             }
 
-            if !suggested {
+            let (potential_candidates, explicitly_negative) = if param_type.is_some() {
+                // Negative bounds are not supported
+                (candidates, Vec::new())
```

Comment:
This is actually sort of a shortcut, consider the following code:

```rust
#![feature(negative_impls)]

trait Foo {
    fn foo(&self);
}

trait Bar {}

impl<T: Foo> !Bar for T {}

fn foo<T: Bar>(x: T) {
    x.foo()
}
```

Which gives the following error:

```
error[E0599]: no method named `foo` found for type parameter `T` in the current scope
  --> main.rs:12:7
   |
12 |     x.foo()
   |       ^^^ method not found in `T`
   |
   = help: items from traits can only be used if the type parameter is bounded by the trait
help: the following trait defines an item `foo`, perhaps you need to restrict type parameter `T` with it:
   |
11 | fn foo<T: Foo + Bar>(x: T) {
   |        ^^^^^^^^

error: aborting due to previous error
```

So I think this could be improved but I think it's reasonable to do it in a follow-up PR since this PR does not change the current diagnostic.